docs/data-flow.md

```diff
@@ -0,0 +1,103 @@
+# Upstream Service Integration
+
+Upstream services (anything that interacts with end users) need to flow their data to `serverless-user-service` and `serverless-erp-service` for data analytics. The specifics of each integration and what data should be provided is to be defined in each service documentation. This page documents the data flow process itself.
+
+## Data Models
+
+At the end of the journey:
+
+- `serverless-user-service` should receive the user data required to generate a user entry
+- `serverless-erp-service` should receive the data required to satisfy the Doc Types defined in ERP Next
+
+Any payload sent to `supporter-event-service` should validate against the `ProducerRequest` found in `data-models`.
+
+The format of the request is the following:
+
+```javascript
+{
+  "user-service": PartialSignupRequest,
+  "erp-service": ErpProducerRequest,
+}
+```
+
+Neither key is required but at least one should be defined.
+
+See:
+- [ProducerRequest](https://github.com/comicrelief/data-models/blob/master/src/schemas/supporter-events/models/requests/ProducerRequest.ts)
+- [Complete examples (mocks)](https://github.com/comicrelief/data-models/blob/master/src/schemas/supporter-events/mocks/requests/ProducerRequest.mock.ts)
+- [PartialSignupRequest](https://github.com/comicrelief/data-models/blob/master/src/schemas/user-service/models/requests/SignupRequest.ts)
+- [ErpProducerRequest](https://github.com/comicrelief/data-models/blob/master/src/schemas/erp/models/requests/ProducerRequest.ts)
+
+### `user-service` key
+
+`serverless-user-service` payloads are meant to be completed by `supporter-event-service`. In particular, `supporter-event-service` will loop through `identities` and hash `identities.content` and add it as `identities.uuid` key if it is not already defined, effectively transforming a `PartialSignupRequest` (accepted by `supporter-event-service`) into a `SignupRequest` (accepted by `serverless-user-service`).
```

Comment:
Is the hashing code in place already? Curious to see it

Reply:
Yes it is:
https://github.com/comicrelief/supporter-event-service/blob/master/src/utils.js

Reply:
Cool. Is that making a uuid?

Reply:
it is not. And your love for inconsistency is going to be helpful here! Some stuff was defined as UUID originally and deployed, now they are not UUIDs anymore but it is quite an effort to rename them (as now they are columns in tables)

Reply:
Ah haha. Are we dealing with actual data yet? Either way, perhaps now is the time to rename them before we go further down this path?

Reply:
Either way - maybe add a note to explain this here?

Reply:
The downstream expects them to be called `uuid`, it's the name of the column. I don't think there is any data on production but I am not sure whether it is worth the effort

Reply:
Yeah cool maybe just a note (it's called uuid for historical reasons but not necessarily a uuid) or something

Reply:
> Either way - maybe add a note to explain this here?

That's an implementation detail, I don't think it should live here. We might decide to fix this. It's not really important how these things are named because they are handled by `supporter-event-service` and `serverless-user-service`

Reply:
Ok, it's just I read this and thought 'that doesn't sound right, have they made a mistake' is all